### ISSUE TITLE
[v13.4.1] Error on submission of Material Request and Delivery Note

### ISSUE THREAD

**Issue by manan-ft** [frappe/erpnext#39660]
## Description of the issue
Getting error `TypeError: 'NoneType' object is not iterable` on Submission of Delivery Note and Material Request

## Context information (for bug reports)
Getting this error only after upgrading to v13.4.1
Has something to do with Budget. I do not have any budgets set

**Output of `bench version`**
```
erpnext 13.4.1
frappe 13.4.1

```

## Steps to reproduce the issue

1. Create a Material Request or Delivery Note
2. Submit the Delivery Note

### Observed result
Server Error: TypeError: 'NoneType' object is not iterable

### Expected result
Submission of document without any error

### Stacktrace / full error message

```
Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/app.py", line 68, in application
    response = frappe.api.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/api.py", line 54, in handle
    return frappe.handler.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 31, in handle
    data = execute_cmd(cmd)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 67, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1175, in call
    return fn(*args, **newargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/desk/form/save.py", line 19, in savedocs
    doc.submit()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 928, in submit
    self._submit()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 917, in _submit
    self.save()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 284, in save
    return self._save(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 336, in _save
    self.run_post_save_methods()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 996, in run_post_save_methods
    self.run_method("on_submit")
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 858, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 1147, in composer
    return composed(self, method, *args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 1130, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 852, in <lambda>
    fn = lambda self, *args, **kwargs: getattr(self, method)(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/erpnext/erpnext/stock/doctype/material_request/material_request.py", line 95, in on_submit
    self.validate_budget()
  File "/home/frappe/frappe-bench/apps/erpnext/erpnext/controllers/buying_controller.py", line 712, in validate_budget
    validate_expense_against_budget(args)
  File "/home/frappe/frappe-bench/apps/erpnext/erpnext/accounts/doctype/budget/budget.py", line 112, in validate_expense_against_budget
    lft, rgt = frappe.db.get_value(doctype, args.get(budget_against), ["lft", "rgt"])
TypeError: 'NoneType' object is not iterable
```

## Additional information

OS version / distribution, `Frappe` install method, etc.

**Comment by Nihantra-Patel** [comment]
Hi @manan-ft,

**We no longer provide support for version 13 or older.**

Kindly upgrade to version 14 or 15, then refresh and check again because lots of bugs fixed in the new version.

Everything is working fine. No problems were encountered in the latest versions 14 and 15.
If the issue occurs then please create a new issue according to the latest version (14 or 15) with a video document if possible.

Version | EOL | Branch
-- | -- | --
Version 12 | End of 2022 | version-12
Version 13 | End of 2023 | version-13
Version 14 | End of 2024 (planned) | version-14
Version 15 | End of 2026 (planned) | version-15
Bleeding edge | N/A | develop


For more information, please refer to the [guidelines](https://github.com/frappe/erpnext/wiki/Supported-Versions).

Thank You!
